src/controller/python/ChipDeviceController-ScriptBinding.cpp

```diff
@@ -455,28 +453,10 @@ ChipError::StorageType pychip_DeviceController_SetWiFiCredentials(const char * s
 
 ChipError::StorageType pychip_DeviceController_CloseSession(chip::Controller::DeviceCommissioner * devCtrl, chip::NodeId nodeid)
 {
-#if 0
-    //
-    // Since we permit multiple controllers per fabric and each is associated with a unique fabric index, closing a session
-    // requires us to do so across all controllers on the same logical fabric.
-    //
-    // TODO: Enable this and remove the call below to DisconnectDevice once #19259 is completed. This is because
-    //       OperationalDeviceProxy instances that are currently active will remain un-affected by this call and still
-    //       provide a valid SessionHandle in the OnDeviceConnected call later when we call DeviceController::GetConnectedDevice.
-    //       However, it provides a SessionHandle that is incapable of actually vending exchanges since it is in a defunct state.
-    //
-    //       For now, calling DisconnectDevice will at least just correctly de-activate a currently active OperationalDeviceProxy
-    //       instance and ensure that subsequent attempts to acquire one will correctly re-establish CASE on the fabric associated
-    //       with the provided devCtrl.
-    //
-    auto err = devCtrl->SessionMgr()->ForEachCollidingSession(ScopedNodeId(nodeid, devCtrl->GetFabricIndex()), [](auto *session) {
-        session->MarkAsDefunct();
-    });
+    devCtrl->SessionMgr()->ForEachMatchingSession(ScopedNodeId(nodeid, devCtrl->GetFabricIndex()),
```

Comment:
Is there any API that can be done to cause a `CLOSE_SESSION` status report attempt as well? MarkAsDefucnt doesn't do this.

@mrjerryjohns ^^^

Reply:
I am not following what you mean by `CLOSE_SESSION` status report?

Reply:
Ah... From the spec: `4.10.1.1. Session Establishment - Out of Resources`

I would argue that this was always an issue even before what is done here in this PR. I recommend that we create a separate issue for that and fix that in a follow up PR

Reply:
Agreed that that is a separable concern.

A session is marked defunct when we did not get back a response in time on an exchange and we think the session on the other side MAY have disappeared. However, there is a possibility that isn't true, and we might still get messages back on that session. That is the material difference between defunct and expired (in the latter case, we are actually forcibly removing the session from our tables because we don't expect any more messages EVER on that session).

So in the defunct case, we _should not_ be sending out a CLOSE_SESSION, because we don't actually want to remove that session on the other side.

Reply:
We should be sending out a CLOSE_SESSION however when expiring sessions.

Reply:
Created https://github.com/project-chip/connectedhomeip/issues/21894 to do that in a follow up PR